src/Imgix/UrlBuilder.cs

```diff
@@ -162,35 +269,104 @@ private String
+        /// then the list begins where it ends.
+        /// 
+        /// When the `while` loop terminates, `begin` is greater than `end`
+        /// (where `end` less than or equal to `MAX_WIDTH`). This means that
+        /// the most recently appended value may, or may not, be the `end`
+        /// value.
+        /// 
+        /// To be inclusive of the ending value, we check for this case and the
+        /// value is added if necessary.
+        /// </summary>
+        /// <param name="begin">beginning (min) image width value</param>
+        /// <param name="end">ending (max) image width value</param>
+        /// <param name="tol">tolerable amount of width value variation, 1-100.</param>
+        /// <returns>list of image width values</returns>        
         public static List<int>
             GenerateTargetWidths(
-                double start = 100, double stop = 8192, double tol = 8)
+                int begin = MinWidth,
+                int end = MaxWidth,
+                int tol = SrcSetWidthTolerance)
+        {
+            return ComputeTargetWidths(begin, end, tol);
+        }
+
+        public static List<int> GenerateEvenTargetWidths()
+        {
+            return SrcSetTargetWidths.ToList();
+        }
+
+        /**
+         * Create an `ArrayList` of integer target widths.
+         *
+         * This function is the implementation details of `targetWidths`.
+         * This function exists to provide a consistent interface for
+         * callers of `targetWidths`.
+         *
+         * This function implements the syntax that fulfills the semantics
+         * of `targetWidths`. Meaning, `begin`, `end`, and `tol` are
+         * to be whole integers, but computation requires `double`s. This
+         * function hides this detail from callers.
+         */
+        private static List<int> ComputeTargetWidths(double begin, double end, double tol)
         {
-            int MAX_SIZE = 8192;
+            if(NotCustom(begin, end, tol))
+            {
+                // If not custom, return the default target widths.
+                return SrcSetTargetWidths.ToList();
+            }
 
-            if (start == stop)
+            if (begin == end)
             {
-                return new List<int> {MakeEven(start)};
+                return new List<int> {(int) Math.Round(begin) };
             }
 
             List<int> resolutions = new List<int>();
-            while (start < stop && start < MAX_SIZE)
+            while (begin < end && begin < MaxWidth)
             {
-                resolutions.Add(MakeEven(start));
-                start *= 1 + (tol / 100) * 2;
+                resolutions.Add((int) Math.Round(begin));
+                begin *= 1 + (tol / 100) * 2;
             }
 
-            if (resolutions.Last() < stop)
+            if (resolutions.Last() < end)
             {
-                resolutions.Add(MakeEven(stop));
+                resolutions.Add((int) Math.Round(end));
```

Comment:
Question: in a situation where `end` > `begin`, what does `ComputeTargetWidths` return?